connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/AiAgentTests.java

```diff
@@ -0,0 +1,439 @@
+import dev.langchain4j.data.message.ChatMessageDeserializer;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.ToolExecutionResultMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.chat.response.ChatResponseMetadata;
+import dev.langchain4j.model.output.FinishReason;
+import dev.langchain4j.model.output.TokenUsage;
+import io.camunda.client.api.search.response.Incident;
+import io.camunda.client.api.worker.JobWorker;
+import io.camunda.connector.agenticai.aiagent.memory.AgentContextChatMemoryStore;
+import io.camunda.connector.agenticai.aiagent.model.AgentContext;
+import io.camunda.connector.agenticai.aiagent.model.AgentMetrics;
+import io.camunda.connector.agenticai.aiagent.model.AgentResponse;
+import io.camunda.connector.agenticai.aiagent.model.AgentState;
+import io.camunda.connector.agenticai.aiagent.provider.ChatModelFactory;
+import io.camunda.connector.e2e.ZeebeTest;
+import io.camunda.process.test.impl.assertions.CamundaDataSource;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import org.assertj.core.api.ThrowingConsumer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.springframework.test.context.bean.override.mockito.MockitoBean;
+
+public class AiAgentTests extends BaseAgenticAiTest {
+
+  @MockitoBean private ChatModelFactory chatModelFactory;
+  @Mock private ChatLanguageModel chatModel;
+  @Captor private ArgumentCaptor<ChatRequest> chatRequestCaptor;
+
+  private JobWorker jobWorker;
+  private final AtomicInteger jobWorkerCounter = new AtomicInteger(0);
+  private final AtomicReference<Map<String, Object>> userFeedbackVariables =
+      new AtomicReference<>(Collections.emptyMap());
+
+  @BeforeEach
+  void setUp() {
+    when(chatModelFactory.createChatModel(any())).thenReturn(chatModel);
+
+    userFeedbackVariables.set(Collections.emptyMap());
+    jobWorkerCounter.set(0);
+
+    jobWorker =
+        camundaClient
+            .newWorker()
+            .jobType("user_feedback")
+            .handler(
+                (client, job) -> {
+                  jobWorkerCounter.incrementAndGet();
+                  client
+                      .newCompleteCommand(job.getKey())
+                      .variables(userFeedbackVariables.get())
+                      .send()
+                      .join();
+                })
+            .open();
```

Comment:
I struggled to complete the service task for "user feedback" within the test case/via API, so I created this worker which sets variables based on what is set in the `mockChatInteractions` setup. Happy to change this to a more concise method if possible.